event/events.go

```diff
@@ -47,12 +54,21 @@ func Multiplex(events ...EventEmitter) *multiplexedEvents {
 
 // The events struct has methods for working with events.
 type events struct {
-	eventSwitch *evts.EventSwitch
+	eventSwitch *go_events.EventSwitch
 }
 
 // Subscribe to an event.
-func (this *events) Subscribe(subId, event string, callback func(evts.EventData)) error {
-	this.eventSwitch.AddListenerForEvent(subId, event, callback)
+func (this *events) Subscribe(subId, event string,
+	callback func(txs.EventData)) error {
+	cb := func(evt go_events.EventData) {
+		eventData, err := mapToOurEventData(evt)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to map go-events EventData to our EventData %v",
+				err))
```

Comment:
This probably rings alarm bells, but I don't have anywhere sensible to send an Error here. Really it should happen if we have an up-to-date mapping against the tendermint version we are using.

Maybe it's better to drop events on the floor that we don't understand?